spatialpandas/geometry/base.py

```diff
@@ -0,0 +1,640 @@
+        return from_shapely([el.to_shapely() for el in self])
+
+    def __arrow_array__(self, type=None):
+        return self.data
+
+    @classmethod
+    def __from_arrow__(cls, data):
+        return cls(data)
+
+    # Constructor
+    def __init__(self, array, dtype=None, copy=None):
+        # Choose default dtype for empty arrays
+        try:
+            if len(array) == 0 and dtype is None:
+                dtype = 'float64'
+        except:
+            # len failed
+            pass
+
+        if isinstance(dtype, GeometryDtype):
+            # Use arrow type as-is
+            arrow_dtype = dtype.arrow_dtype
+        elif dtype is not None and dtype != np.dtype('object'):
+            # Scalar element dtype
+            arrow_dtype = pa.from_numpy_dtype(dtype)
+
+            # Wrap dtype with appropriate number of nesting levels
+            for i in range(self._nesting_levels):
+                arrow_dtype = pa.list_(arrow_dtype)
+        else:
+            # Let arrow infer type
+            arrow_dtype = None
+
+        # Unwrap Geometry elements to numpy arrays
+        if is_array_like(array) or isinstance(array, list):
+            array = [_unwrap_geometry(el, self._element_type) for el in array]
+            self.data = pa.array(array, type=arrow_dtype)
+        elif isinstance(array, pa.Array):
+            self.data = array
+        elif isinstance(array, pa.ChunkedArray):
+            self.data = pa.concat_arrays(array)
+        else:
+            raise ValueError(
+                "Unsupported type passed for {}: {}".format(
+                    self.__class__.__name__, type(array)
+                )
+            )
+
+        self.offsets = np.array([0])
+
+        # Check that inferred type has the right number of nested levels
+        pyarrow_element_type = _validate_nested_arrow_type(
+            self._nesting_levels, self.data.type
+        )
+
+        self._pyarrow_element_type = pyarrow_element_type
+        self._numpy_element_type = pyarrow_element_type.to_pandas_dtype()
+        self._dtype = self._dtype_class(self._numpy_element_type)
+
+        # Validate input data is compatible
+        offsets = self.buffer_offsets
+
+        # Validate even number of inner elements per polygon
+        if any((offsets[-1] % 2) > 0):
+            raise ValueError("""
+Geometry objects are represented by interleaved x and y coordinates, so they must have
+an even number of elements. Received specification with an odd number of elements.""")
+
+    @property
+    def _dtype_class(self):
+        return GeometryDtype
+
+    @property
+    def dtype(self):
+        return self._dtype
+
+    @property
+    def nbytes(self):
+        size = 0
+        for buf in self.data.buffers():
+            if buf is not None:
+                size += buf.size
+        return size
+
+    nbytes.__doc__ = ExtensionArray.nbytes.__doc__
+
+    def astype(self, dtype, copy=True):
+        if self.dtype == dtype:
+            return self
```

Comment:
Probably a nitpick in practice, but you should do something like `self.copy() if copy else self`

Reply:
:+1: done in 080d7a25e926464b1bc3e9c0022f1f72ee1bb557